src/main/groovy/com/twcable/grabbit/GrabbitConfiguration.groovy

```diff
@@ -117,6 +119,31 @@ class GrabbitConfiguration {
         )
     }
 
+    /**
+     * Removes '/.' from {@code pathConfigurations}'s path if exists in the end, and Sorts the {@code pathConfigurations}
+     * lexicographically on the basis of paths
+     *
+     * @param pathConfigurations
+     *          Collection of {@link PathConfiguration} to sort
+     * @return
+     *          Sorted Collection of {@link PathConfiguration}
+     */
+    private static Collection<PathConfiguration> sortPathConfigs(
+            @Nonnull final Collection<PathConfiguration> pathConfigurations) {
+        Collection<PathConfiguration> modifiablePathConfig = pathConfigurations;
+
+        modifiablePathConfig.collect { element ->
```

Comment:
When you do "Collection<PathConfiguration> modifiablePathConfig = pathConfigurations" you are just creating a new pointer to the same memory; not creating a copy, so modifiablePathConfig.sort() is still acting on the same memory of the argument. Even if you were to do a clone(), that would just clone the Collection object; The references to objects within the collection would stay the same.